### ISSUE TITLE
Make XChange compatible with jakarta.ws.rs-api 3.x

### ISSUE THREAD

**Issue by ppalaga** [knowm/XChange#4676]
As far as I can see, the `jakarta.ws.rs-api` is pulled transitively via `com.github.mmazi:rescu`. Once a `com.github.mmazi:rescu` release compatible with jakarta.ws.rs-api 3.x is available, XChange should be adjusted as well. This is currently blocked by https://github.com/mmazi/rescu/issues/142

It would be nice to have this to be able to use XChange on Quarkus 3.x.

**Comment by martinformi** [comment]
@ppalaga Peter, use this if needed - https://github.com/martinformi/XChange/tree/feature/jakarta-namespace

**Comment by ppalaga** [comment]
> @ppalaga Peter, use this if needed - https://github.com/martinformi/XChange/tree/feature/jakarta-namespace

Great, would you mind sending a pull request against https://github.com/knowm/XChange ?

**Comment by ppalaga** [comment]
Thanks for the PR. It is up to the project maintainers to decide how they deal with the incompatibility. I'd personally be fine with bumping the major version as having both xchange and xchange-jakarta dependencies in one project would not work anyway - it would not be possible to have transitives compatible with both of them.

**Comment by bigscoop** [comment]
@ppalaga looks like the MR https://github.com/knowm/XChange/pull/4735 resolves it. Feel free to test and eventually close the issue

**Comment by jdlee726** [comment]
is already fixed?

**Comment by bigscoop** [comment]
yes, issue can be closed
